scanpy/tests/test_highly_variable_genes.py

```diff
@@ -16,32 +17,39 @@
 FILE_V3_BATCH = Path(__file__).parent / Path("_scripts/seurat_hvg_v3_batch.csv")
 
 
-def test_highly_variable_genes_basic():
+def test_highly_variable_genes_runs():
     adata = sc.datasets.blobs()
     sc.pp.highly_variable_genes(adata)
 
+
+def test_highly_variable_genes_supports_batch():
     adata = sc.datasets.blobs()
-    np.random.seed(0)
-    adata.obs["batch"] = np.random.binomial(3, 0.5, size=(adata.n_obs))
-    adata.obs["batch"] = adata.obs["batch"].astype("category")
+    gen = np.random.default_rng(0)
+    adata.obs["batch"] = pd.array(
+        gen.binomial(3, 0.5, size=adata.n_obs), dtype="category"
+    )
```

Comment:
`np.random.*` functions are deprecated in favor of `np.random.Generator.*` methods.